docs/deployment/resource-providers/standalone/index.md

```diff
@@ -24,153 +24,206 @@ specific language governing permissions and limitations
+
+The `bin/jobmanager.sh` and `bin/taskmanager.sh` scripts support starting the respective daemon in the background (using the `start` argument), or in the foreground (using `start-foreground`). In the foreground mode, the logs are printed to standard out. This mode is useful for deployment scenarios where another process is controlling the Flink daemon (e.g. Docker).
+
+The scripts can be called multiple times, for example if multiple TaskManagers are needed. The instances are tracked by the scripts, and can be stopped one-by-one (using `stop`) or all together (using `stop-all`).
+
+#### Windows Cygwin Users
+
+If you are installing Flink from the git repository and you are using the Windows git shell, Cygwin can produce a failure similar to this one:
 
 {% highlight bash %}
-bin/taskmanager.sh start|start-foreground|stop|stop-all
+c:/flink/bin/start-cluster.sh: line 30: $'\r': command not found
 {% endhighlight %}
 
-Make sure to call these scripts on the hosts on which you want to start/stop the respective instance.
+This error occurs because git is automatically transforming UNIX line endings to Windows style line endings when running on Windows. The problem is that Cygwin can only deal with UNIX style line endings. The solution is to adjust the Cygwin settings to deal with the correct line endings by following these three steps:
+
+1. Start a Cygwin shell.
+
+2. Determine your home directory by entering
 
-## High-Availability with Standalone
+    ```bash
+    cd; pwd
+    ```
+
+    This will return a path under the Cygwin root path.
+
+3. Using NotePad, WordPad or a different text editor open the file `.bash_profile` in the home directory and append the following (if the file does not exist you will have to create it):
+
+    ```bash
+    $ export SHELLOPTS
+    $ set -o igncr
+    ```
+
+4. Save the file and open a new bash shell.
+
+### Setting up High-Availability
 
 In order to enable HA for a standalone cluster, you have to use the [ZooKeeper HA services]({% link deployment/ha/zookeeper_ha.md %}).
 
 Additionally, you have to configure your cluster to start multiple JobManagers.
 
-### Masters File (masters)
-
 In order to start an HA-cluster configure the *masters* file in `conf/masters`:
 
 - **masters file**: The *masters file* contains all hosts, on which JobManagers are started, and the ports to which the web user interface binds.
 
   <pre>
-jobManagerAddress1:webUIPort1
+master1:webUIPort1
 [...]
-jobManagerAddressX:webUIPortX
+masterX:webUIPortX
   </pre>
 
 By default, the job manager will pick a *random port* for inter process communication. You can change this via the [high-availability.jobmanager.port]({% link deployment/config.md %}#high-availability-jobmanager-port) key. This key accepts single ports (e.g. `50010`), ranges (`50000-50025`), or a combination of both (`50010,50011,50020-50025,50050-50075`).
 
-### Example: Standalone Cluster with 2 JobManagers
+#### Example: Standalone HA Cluster with 2 JobManagers
```

Comment:
Can we switch from `<pre/>` tags to "\`\`\`sh \`\`\`"? This way, we would enable syntax highlighting for these code blocks.